ci/destroy_folder.sh

```diff
@@ -0,0 +1,69 @@
+#!/bin/bash
+
+set -o nounset
+set -o errexit
+set -o pipefail
+
+trap 'CHILDREN=$(jobs -p); if test -n "${CHILDREN}"; then kill ${CHILDREN} && wait; fi' TERM
+
+function error_exit() {
+    msg=${1}
+    >&2 echo "[Error] ${msg}"
+    exit 1
+}
+
+function prow_archive_state() {
+    if [ -n "${OPENSHIFT_CI:-}" ]; then
+        echo
+        echo "[INFO] Running in OpenShift CI (Prow)."
+        echo "[INFO] Archiving state for PROW."
+        echo
+        set -o xtrace
+        tar cvfz "${STATE_ARCHIVE}" ./*.tfstate*  #save for later terraform destroy
+        set +o xtrace
+        echo "[INFO] ${STATE_ARCHIVE} created."
+    fi
+
+    if [ -n "${1:-}" ]; then
+        error_exit "TF command failed."
+    fi
+}
+
+THIS_DIR="$( cd "$( dirname "${BASH_SOURCE[0]}" )" >/dev/null 2>&1 && pwd )"
```

Comment:
Also, can all those parameter initialization be in a common function on a different file?